python_tests/pytest_input_generators.py

```diff
@@ -1233,6 +1233,58 @@ def slice_error_generator(
         yield SampleInput(input_tensor, **es.kwargs), es.ex_type, es.ex_str
 
 
+def squeeze_generator(
+    op: OpInfo, dtype: torch.dtype, requires_grad: bool = False, **kwargs
+):
+    make_arg = partial(
+        make_tensor, device="cuda", dtype=dtype, requires_grad=requires_grad
+    )
+
+    # shape, squeeze_dims
+    cases = (
+        ((5, 1, 1), (1, 2)),
+        ((5, 1, 1), (-2, -1)),
+        ((5, 1, 1), (2, 1)),
+        ((5, 1, 1), (-1, -2)),
+        ((1, 5, 1), (0, 2)),
+        ((1, 5, 1), (-3, -1)),
+        ((1, 1, 5), (0, 1)),
+        ((1, 1, 5), (-3, -2)),
+        ((5, 5, 5), ()),
+        ((1, 1, 1), ()),
+        ((1, 1, 1), (0, 1, 2)),
+        ((1, 1, 1), (-3, -2, -1)),
+    )
```

Comment:
can we add a couple test cases:
1. squeeze a non-size-1 dimension.
2. squeeze an expanded dimension. We'll need to adjust the test setup.

Reply:
This may not be obvious but the error checking in the `pytest` tests are only for the Python API.  Therefore, the testing just makes sure the checks in the python API are working as expected given the fusion is not actually run.  The checks that exist based on the Fusion IR would be checked by a C++ test, at this time.   The reason this is the case, is that the op is getting validated in isolation since it is hard to generate a valid Fusion with an invalid Python API usage.  This is why the check for trying to squeeze a dimension that is not 1 is not done here.

Reply:
IIUC, the statement above only applies to `error checking`.

`squeeze a dimension that is not 1` is NOT an error though and we should be able to execute it. I believe our implementation here isn't doing the right thing, hence my suggestion to fix that and add a corresponding test here.